resources/assets/js/annotations/ol/PolygonValidator.js

```diff
@@ -134,6 +130,54 @@ function jstsAddLineString(lineString, polygonizer) {
     polygonizer.add(toAdd);
 }
 
+/**
+ * Removes duplicated subpolygon's coordinates which can be still included in some polygons.
+ *
+ * @param polygons List of (multi)polygons
+ * @returns List of separated polygons
+ *
+ * **/
+function polishPolygons(polygons) {
```

Comment:
I don't really understand this function. The comment says that duplicated coordinates should be removed. I understand this as an operation like `[[x1, y1],[x2,y2],[x2,y2]] -> [[x1, y1],[x2,y2]]`. Also, I think that the multi-polygon here is the list of polygons that you get when a non-simple polygon is cut into simple pieces. So why are all sub-polygons compared here? Shouldn't duplicate coordinates be removed independently for each sub-polygon? And as I said before, can't you determine the biggest sub-polygon first before performing all these operations (just asking, maybe you *really* can't)?

Since I struggled to understand this, maybe more elaborate comments should be added (for future readers). An example of the special case that this fixes would also be useful (since this function wasn't there before and it already looked like it would work fine).

Reply:
Ah I missed your comment above. So determining the biggest polygon is not possible before converting back to OL features. But the questions about the new function are still valid.

Reply:
Maybe the "polishing" can be done only on the biggest polygon?

Reply:
> I understand this as an operation like `[[x1, y1],[x2,y2],[x2,y2]] -> [[x1, y1],[x2,y2]]`

What I meant was something like this: (original) polygon1 = [[x1,y1], [x2,y2]] and and (subpolygon) polygon2 = [[x2,y2]]. 
The polygon was separated but the coordinates were not removed from the original polygon "polygon1".
This case can occur when using the magic wand tool and modifying that polygon.

Reply:
> Shouldn't duplicate coordinates be removed independently for each sub-polygon

Unfortunately, no. Like I described above, the duplicates occur between different polygons and not within one.

Reply:
> Maybe the "polishing" can be done only on the biggest polygon?

The polish method is used to create "clean" polygons whose coordinate sets occur only once. Therefore I need to compare the polygon with the most coordinate sets with the others to check if one is contained in another.

Reply:
Ok thanks for the explanation! I think I'll tinker with this a bit once you added the comments about "polishing", so I better understand what's going on :wink:

Reply:
I don't know if it helps, but I recorded the bug which I fixed with the polish method. It doesn't happen all the time when modifying a magic wand polygon, but there are cases which you can see in the video.


https://github.com/biigle/core/assets/72259845/2af9349d-3ffd-4728-95b3-a45c6c139cc1